File: resource-registry/src/test/java/nl/haploid/octowight/registry/repository/ResourceModelDmoFactoryTest.java
package nl.haploid.octowight.registry.repository;

import mockit.Injectable;
import mockit.Mocked;
import mockit.StrictExpectations;
import mockit.Tested;
import nl.haploid.octowight.registry.TestData;
import nl.haploid.octowight.registry.data.Resource;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class ResourceModelDmoFactoryTest {

	@Tested
	private ResourceModelDmoFactory resourceModelDmoFactory;

	@Injectable
	private ResourceModelIdFactory resourceModelIdFactory;

	@Test
	public void testFromResourceAndBody(final @Mocked Resource resource) {
		final ResourceModelId resourceModelId = TestData.resourceModelId();
		final String body = TestData.nextString();
		final Long version = TestData.nextLong();
		new StrictExpectations() {{
			resourceModelIdFactory.resourceModelId(resource);
			times = 1;
			result = resourceModelId;
			resource.getVersion();
			times = 1;
			result = version;
		}};
		final ResourceModelDmo actualResourceModelDmo = resourceModelDmoFactory.fromResourceAndBody(resource, body);
		assertEquals(resourceModelId, actualResourceModelDmo.getId());
		assertEquals(body, actualResourceModelDmo.getBody());
		assertEquals(version, actualResourceModelDmo.getVersion());
	}
}

File: resource-registry/src/test/scala/nl/haploid/octowight/registry/repository/ResourceModelDmoFactoryTest.scala
package nl.haploid.octowight.registry.repository

import nl.haploid.octowight.registry.data.Resource
import nl.haploid.octowight.registry.{AbstractTest, TestData}
import nl.haploid.octowight.{Mocked, Tested}

class ResourceModelDmoFactoryTest extends AbstractTest {
  @Tested private val resourceModelDmoFactory = new ResourceModelDmoFactory
  @Mocked private val resourceModelIdFactory: ResourceModelIdFactory = null

  "Resource model DMO factory" should "build DMO from resource" in {
    val resource = mock[Resource[_]]
    val resourceModelId = TestData.resourceModelId
    val body = TestData.nextString
    val version = TestData.nextLong
    expecting {
      resourceModelIdFactory.resourceModelId(resource) andReturn resourceModelId once()
      resource.getVersion andReturn version once()
    }
    whenExecuting(resourceModelIdFactory, resource) {
      val actualResourceModelDmo = resourceModelDmoFactory.fromResourceAndBody(resource, body)
      actualResourceModelDmo.getId should be(resourceModelId)
      actualResourceModelDmo.getBody should be(body)
      actualResourceModelDmo.getVersion should be(version)
    }
  }
}
